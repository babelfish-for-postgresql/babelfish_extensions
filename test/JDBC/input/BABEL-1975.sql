EXEC sp_babelfish_configure 'babelfishpg_tsql.escape_hatch_session_settings', 'strict';
GO

DROP TABLE if EXISTS t1;
CREATE TABLE t1 (c1 int);
INSERT INTO t1 values (1);
INSERT INTO t1 values (2);
INSERT INTO t1 values (NULL);
GO

SELECT * FROM t1;
SELECT @@rowcount;
GO

SET ROWCOUNT 0;
SELECT @@rowcount;
GO

SELECT * FROM t1;
SELECT @@rowcount;
GO

-- test invalid settings
SET ROWCOUNT 1;
SELECT @@rowcount;
GO

set ROWCOUNT 0;
SELECT @@rowcount;
GO

-- clean up
DROP TABLE t1;
GO

-- reset to default
EXEC sp_babelfish_configure 'babelfishpg_tsql.escape_hatch_session_settings', 'ignore';
GO
