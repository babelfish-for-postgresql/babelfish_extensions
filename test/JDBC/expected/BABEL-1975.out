EXEC sp_babelfish_configure 'babelfishpg_tsql.escape_hatch_session_settings', 'strict';
GO

DROP TABLE if EXISTS t1;
CREATE TABLE t1 (c1 int);
INSERT INTO t1 values (1);
INSERT INTO t1 values (2);
INSERT INTO t1 values (NULL);
GO
~~ROW COUNT: 1~~

~~ROW COUNT: 1~~

~~ROW COUNT: 1~~


SELECT * FROM t1;
SELECT @@rowcount;
GO
~~START~~
int
1
2
<NULL>
~~END~~

~~START~~
int
3
~~END~~


SET ROWCOUNT 0;
SELECT @@rowcount;
GO
~~START~~
int
0
~~END~~


SELECT * FROM t1;
SELECT @@rowcount;
GO
~~START~~
int
1
2
<NULL>
~~END~~

~~START~~
int
3
~~END~~


-- test invalid settings
SET ROWCOUNT 1;
SELECT @@rowcount;
GO
~~START~~
int
0
~~END~~


set ROWCOUNT 0;
SELECT @@rowcount;
GO
~~START~~
int
0
~~END~~


-- clean up
DROP TABLE t1;
GO

-- reset to default
EXEC sp_babelfish_configure 'babelfishpg_tsql.escape_hatch_session_settings', 'ignore';
GO
